src/App.tsx

```diff
@@ -14,11 +14,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isVisible: boolean,
+  goodsList: string[],
+};
+
+class App extends React.Component <{}, State> {
+  state: Readonly <State> = {
+    isVisible: false,
+    goodsList: [...goodsFromServer],
+  };
+
+  visabilityToggle = () => {
+    this.setState(state => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      goodsList: [...state.goodsList].reverse(),
+    }));
```

Comment:
Great job! Your solution is quite good! But there is one thing I'm not sure. Every your action is creating new array. But you alredy created(copied) it in your initial state. Isn't it way better to work with it instead and create the new one only if it's really necessary?